File: LG_ArchJ/app/src/main/java/com/louis/lg_archj/MyApplication.java
package com.louis.lg_archj;

import android.app.Application;

import com.louis.lg_archj.di.AppContainer;

public class MyApplication extends Application {
    @Override
    public void onCreate() {
        super.onCreate();

        // 初始化依赖注入容器
        AppContainer.getInstance().initialize(this);
    }

    @Override
    public void onTerminate() {
        try {
            AppContainer.getInstance().cleanup();
        } catch (Exception e) {
            e.printStackTrace();
        }
        super.onTerminate();
    }
}

File: MyComposeProject/app/src/main/java/com/louis/mycomposeproject/MyApplication.kt
package com.louis.mycomposeproject

import android.app.Application
import com.louis.mycomposeproject.util.Logger

class MyApplication : Application() {
    override fun onCreate() {
        super.onCreate()
        
        // 初始化依赖
        Dependencies.init(applicationContext)
        
        // 设置日志模式
        Logger.setDebugMode(BuildConfig.DEBUG)
        
        Logger.i("Application started")
    }
}
